src/bloqade/pyqrack/device.py

```diff
@@ -0,0 +1,166 @@
+from typing import Any, TypeVar, ParamSpec
+from dataclasses import field, dataclass
+
+import numpy as np
+from kirin import ir
+
+from pyqrack.pauli import Pauli
+from bloqade.device import AbstractSimulatorDevice
+from bloqade.pyqrack.reg import Measurement, PyQrackQubit
+from bloqade.pyqrack.base import (
+    MemoryABC,
+    StackMemory,
+    DynamicMemory,
+    PyQrackOptions,
+    PyQrackInterpreter,
+    _default_pyqrack_args,
+)
+from bloqade.pyqrack.task import PyQrackSimulatorTask
+from bloqade.analysis.address.lattice import AnyAddress
+from bloqade.analysis.address.analysis import AddressAnalysis
+
+RetType = TypeVar("RetType")
+Params = ParamSpec("Params")
+
+
+@dataclass
+class PyQrackSimulatorBase(AbstractSimulatorDevice[PyQrackSimulatorTask]):
+    options: PyQrackOptions = field(default_factory=_default_pyqrack_args)
+    loss_m_result: Measurement = field(default=Measurement.One, kw_only=True)
```

Comment:
I think this part will need some co-design with a target in Kirin. The pattern of running something on a device and wrap it in a task object is probably something we can generalize in Kirin.